### PR TITLE
[Compression] v2.5 stage 4: add compressor

### PR DESCRIPTION
**Pull request by J-shang**
#### Test Options ####
  - [ ] fast test
  - [ ] full test - HPO
  - [ ] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [ ] test case
  - [ ] doc

### CODE REVIEW

File: nni/contrib/compression/base/compressor.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        """
+        if compressor._is_wrapped:
+            compressor.unwrap_model()
+        model = compressor.bound_model
+        existed_wrappers = compressor._module_wrappers
+        if compressor.evaluator is not None and evaluator is not None:
+            _logger.warning('compessor already has evaluator, the new evaluator passed to this function will be ignored.')
+        evaluator = compressor.evaluator if compressor.evaluator else evaluator
+
+        # note that here don't have `mode` because subclass should know what its mode is.
```
Comment (Bonytu) [inline nni/contrib/compression/base/compressor.py:86]: a little confused, where is the mode parameter defined? Can you give an example of a subclass?

File: nni/contrib/compression/base/apply_method.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Microsoft Corporation.
```
Comment (QuanluZhang) [inline nni/contrib/compression/base/apply_method.py:1]: do we still need the folder "contrib"?

File: nni/contrib/compression/base/setting.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        },
+        '_output_': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': 'default',
+            'apply_method': 'clamp_round',
+        }
+    }
+
+    avtivation_setting = {
```
Comment (QuanluZhang) [inline nni/contrib/compression/base/setting.py:169]: activation_setting

File: nni/contrib/compression/base/apply_method.py (hunk 2 above)
Comment (J-shang) [reply nni/contrib/compression/base/apply_method.py:1]: we need `contrib` to hold some experimental feature or tutorial assets, but yes, compression v2.5 should be moved out after development is done.

File: nni/contrib/compression/base/compressor.py (hunk 1 above)
Comment (J-shang) [reply nni/contrib/compression/base/compressor.py:86]: yes, it's confused, I will del `mode` from init interface, instead, add it as a class attr.

File: nni/contrib/compression/base/compressor.py [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+        return cls(model=model, config_list=new_config_list, evaluator=evaluator, existed_wrappers=existed_wrappers, *args, **kwargs)
```
Comment (J-shang) [inline nni/contrib/compression/base/compressor.py:87]: `*args` is wrong, remove it before merge or fix it in another pr

**Review by Bonytu — APPROVED** [review]

File: nni/contrib/compression/base/compressor.py (hunk 4 above)
Comment (J-shang) [reply nni/contrib/compression/base/compressor.py]: fixed

File: nni/contrib/compression/base/setting.py (hunk 3 above)
Comment (J-shang) [reply nni/contrib/compression/base/setting.py]: fixed

**Review by QuanluZhang — APPROVED** [review]

**Pull request merged**